pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -7491,3 +7497,122 @@ func TestRhpamKieserverEnvWithoutDecisionsOnlyEnabled(t *testing.T) {
 	result, _ := strconv.ParseBool(getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DECISIONS_ONLY"))
 	assert.False(t, result)
 }
+
+func TestCredentialWithCredentialSecretAndCredentialsUseAsDefault(t *testing.T) {
```

Comment:
I think it would be good to improve some existing test to make sure the old way still working for smartrouter and dashbuilder.